Src/EB/AMReX_EB2_3D_C.cpp

```diff
@@ -646,18 +671,48 @@ void build_cells (Box const& bx, Array4<EBCellFlag> const& cell,
 
             // remove small cells
             if (vfrac(i,j,k) < small_volfrac) {
-                vfrac(i,j,k) = 0.0;
-                vcent(i,j,k,0) = 0.0;
-                vcent(i,j,k,1) = 0.0;
-                vcent(i,j,k,2) = 0.0;
-                bcent(i,j,k,0) = -1.0;
-                bcent(i,j,k,1) = -1.0;
-                bcent(i,j,k,2) = -1.0;
-                bnorm(i,j,k,0) = 0.0;
-                bnorm(i,j,k,1) = 0.0;
-                bnorm(i,j,k,2) = 0.0;
-                barea(i,j,k) = 0.0;
-                cell(i,j,k).setCovered();
+
+                set_covered(i,j,k,cell,vfrac,vcent,barea,bcent,bnorm);
+                
+                if(extend_domain_face) {
+                   const auto & dlo = domain.loVect();
+                   const auto & dhi = domain.hiVect();
+                   const auto & lo = bxg1.loVect();
+                   const auto & hi = bxg1.hiVect();
+
+                   // if cell lies on any of the 6 domain faces, force
+                   // all layers of corresponding ghost cells to be covered
+                   if(i == dlo[0]) {
+                     for(int ii = lo[0]; ii < dlo[0]; ++ii) {
+                        set_covered(ii,j,k,cell,vfrac,vcent,barea,bcent,bnorm);
```

Comment:
The problem is there is a race condition on GPU.  There is another thread that has been assigned to work on cell (ii,j,k).  If both threads are trying to write the data of cell(ii,j,k), it's undefined which thread will win.  I think you will have to fix the extended face after this kernel.  You can launch a kernel on `bxg1`.  In the kernel, you test on if this cell in extended region.  If yes, you test on its original source to see if that has been covered.  If that is covered, you can then set this extended cell to covered.  Also note that you need to make sure periodic boundary has been taken into account.  We do not extend at periodic boundaries.